crates/kitsune-db/src/pool.rs

```diff
@@ -0,0 +1,47 @@
+use diesel_async::{
+    pooled_connection::deadpool::{Object, Pool, PoolError},
+    scoped_futures::ScopedBoxFuture,
+    AsyncConnection, AsyncPgConnection,
+};
+use std::future::Future;
+
+#[derive(Clone)]
+pub struct PgPool {
+    inner: Pool<AsyncPgConnection>,
+}
+
+impl PgPool {
+    /// Run the code inside a context with a database connection
+    pub async fn with_connection<F, Fut, T, E>(&self, func: F) -> Result<T, E>
```

Comment:
I'm not happy with this API. It always requires a compatible error type.  
That just sucks because every invokation has to get its error mapped into a compatible type, adding a bunch of `.map_err(Error::from)` repetiton

Reply:
Found a nice solution for this. Created a new error enum that wraps the pool error and a user supplied error, we then just need a single `From` impl on the error type